submissions/YuraZagor/tjsw/index.js

```diff
@@ -0,0 +1,166 @@
+const manNamesArr = ['Vasil', 'Yurij', 'Zahar', 'Taras', 'Olexiy', 'Oleg', 'Mykola', 'Sashko', 'Sergij', 'Bohdan'];
+const womanNamesArr = ['Hanna', 'Helena', 'Vasilisa', 'Yulya', 'Oksana', 'Olena', 'Olga', 'Myroslava', 'Svitlana', 'Bohdana'];
+const animalNamesArr = ['Joy', 'Happy', 'Lucky', 'Puffy'];
+const genderArr = ['male', 'female'];
+
+let catSay = 'meau';
+let dogSay = 'woufff';
+let humanSay = 'Glory to Ukraine!';
+
+function randomSelect(arr){
+  return arr[Math.floor(Math.random()*arr.length)]
+};
+
+let manName = randomSelect(manNamesArr);
+let womanName = randomSelect(womanNamesArr);
+let animalName = randomSelect(animalNamesArr);
+
+const inhabitants = [
+  { species: 'human', 
+    gender: 'male',
+    name: 'Vasil',
+    saying: 'Slava natsiyi',
+    friend: 'Hanna',
+    legs: 2,
+    hands: 2,
+  },
+  { species: 'human', 
+    gender: 'female',
+    name: 'Hanna',
+    saying: 'Smert vorogam',
+    friend: 'Vasil',
+    legs: 2,
+    hands: 2,
+  },];
+
+const dogsVolier = [{ 
+  species: 'Dog', 
+  gender: 'female',
+  name: 'Guffy',
+  saying: dogSay,
+  friend: randomSelect(inhabitants).name,
+  paws: 4,
+  tail: 1,
+},];
+
+class Inhabitant {  
+  constructor(species, name, saying, friend, gender, habitat = inhabitants){
```

Comment:
I see that.  I reacted as I could to your advice a bit ago: 
_Current implementation enforces peer developers to have a single one container for inhabitants and it has to be named inhabitants. OOP helps to be flexible and introduce concepts with minimal effect on code.
Better design would allow us to add containers like cities, communities etc - all populated with the same inhabitants (in different combinations). Reiterate over OOP principles if needed and improve this part._ 

Also, I do not see any other way but using global array to fulfil the requirement **OOP Exercise 2.i: Friends list is a list of objects refs to other inhabitants rather than just names (strings)** As I get it, they all need to be pushed into that array, and any of them has to have access to that array. May be I am wrong, but these are my first programs and I've had no expirience to rely on